### DISCUSSION TITLE
Feature Request: Instructions for decrypting an encrypted QR without Krux

### DISCUSSION THREAD

**Discussion by 2oh1** [selfcustody/krux#282] [Ideas]
I absolutely love the idea of using encrypted QRs.  I cannot stress that enough.  What a fantastic feature!  In fact, I wish there was an option to create encrypted passphrase QRs.  But I'd feel safer if I knew how to decrypt an encrypted QR without needing Krux, just in case Krux ever ceases to be, say a decade from now.

To be clear: I don't actually want to decrypt QRs outside of Krux.  I just want to be able to prove to myself that I can, so I can feel safer using encrypted QRs.

**Comment by odudex** [comment]
Since the begging I promised a python script for that. I'll do it. We can also add it as a feature to other utility GUIs @qlrd is working on.
For now, there's also the [Android app](https://github.com/odudex/krux_binaries/tree/main/Android) which can be used to decrypt the QRs and you could also run and use the Krux simulator on PC. But let's leave this issue here until we have more simple options for PC.

**Comment by jdlcdl** [comment]
Till then, some links to references:
* [explanation in docs](https://github.com/selfcustody/krux/blob/main/docs/getting-started/features/encrypted-mnemonics.en.md)
* [the code itself](https://github.com/selfcustody/krux/blob/main/src/krux/encryption.py)
* [an example decrypting from flash](https://github.com/selfcustody/k210comb/blob/master/docs/hex_dump_example.md#previously-the-spi-flash-file-system-changed)

And the sane consideration that "Encrypted seed is nice for daily use, but since there is a 'key', it's not sufficient as the only seed backup".

**Comment by odudex** [comment]
Done:
[python script](https://gist.github.com/odudex/e86fb4d97cc2c161803ad01f7b43e237)

Python requirements: pycryptodome, opencv-python, pyzbar, embit
`pip install pycryptodome opencv-python pyzbar embit`
OS may require zbar-tools to scan QRs
`sudo apt install zbar-tools`

Usage:
Decrypt a mnemonic from a json file (mnemonic stored on SD card):
`python krux_decrypt.py json --file seeds.json`
<seeds.json> can be a path if the file is not on the same folder as the script

Scan and decrypt an encrypted mnemonic QR code:
`python krux_decrypt.py qr`

**Reply by 2oh1** [reply]
Thanks!!!  Krux is such a fantastic project.
